http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -669,6 +669,80 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
+  // 1. The request includes a threshold which is invalid
+  // 2. The request includes enabled=true but no threshold is provided, and the existing DB threshold is invalid
+  private def validateUpdateAppConfigRequest(req: UpdateAppConfigRequest, dbApp: App)(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] = for {
+    ctx <- as.ask
+
+    // 1. The request includes a threshold which is invalid
+    invalidThresholdRequest = req.autodeleteThreshold.exists(_ <= 0)
+    _ <- F.raiseWhen(invalidThresholdRequest)(
+      BadRequestException("invalid value for autodeleteThreshold", Some(ctx.traceId))
+    )
+
+    // 2. The request includes enabled=true but no threshold is provided, and the existing DB threshold is invalid
+    wantEnableWithoutThresholdButDbInvalid = req.autodeleteEnabled.getOrElse(false) &&
+      req.autodeleteThreshold.isEmpty &&
+      dbApp.autodeleteThreshold.getOrElse(0) <= 0
+
+    _ <- F.raiseWhen(wantEnableWithoutThresholdButDbInvalid)(
+      BadRequestException(
+        "when enabling autodelete without passing an explicit autodeleteThreshold, the existing config must be valid",
+        Some(ctx.traceId)
+      )
+    )
+  } yield ()
+
+  private def updateAppConfigInternal(appId: AppId, validatedChanges: UpdateAppConfigRequest): F[Unit] = for {
+    _ <- validatedChanges.autodeleteEnabled.fold(F.unit)(enabled =>
+      appQuery
+        .updateAutodeleteEnabled(appId, enabled)
+        .transaction
+        .void
+    )
+
+    // note: does not clear the threshold if None.  This only sets defined thresholds.
+    _ <- validatedChanges.autodeleteThreshold.fold(F.unit)(threshold =>
+      appQuery
+        .updateAutodeleteThreshold(appId, Some(threshold))
+        .transaction
+        .void
+    )
+  } yield ()
+
+  override def updateAppConfig(userInfo: UserInfo,
+                               cloudContext: CloudContext.Gcp,
+                               appName: AppName,
+                               req: UpdateAppConfigRequest
+  )(implicit as: Ask[F, AppContext]): F[Unit] =
+    for {
+      ctx <- as.ask
+
+      // throw 403 if no project-level permission
+      hasProjectPermission <- authProvider.isUserProjectReader(cloudContext, userInfo)
+      _ <- F.raiseWhen(!hasProjectPermission)(ForbiddenError(userInfo.userEmail, Some(ctx.traceId)))
+
+      appOpt <- KubernetesServiceDbQueries.getActiveFullAppByName(cloudContext, appName).transaction
+      appResult <- F.fromOption(appOpt,
+                                AppNotFoundException(cloudContext, appName, ctx.traceId, "No active app found in DB")
+      )
+
+      _ <- metrics.incrementCounter("updateAppConfig", 1, Map("appType" -> appResult.app.appType.toString))
+
+      // throw 404 if no UpdateApp permission
+      listOfPermissions <- authProvider.getActions(appResult.app.samResourceId, userInfo)
+      hasPermission = listOfPermissions.toSet.contains(AppAction.UpdateApp)
```

Comment:
Is this pre-existing UpdateApp action in Sam doing what you would expect it to do in terms of permissions?

AKA, how is a user granted this action from Sam?

Reply:
That's a good question.  I don't have a good understanding of how this permission works, or what permission check would be appropriate here.  I will think about that.

Reply:
The list of App Actions seems to match that in Sam: https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf

On L829 I see that a user with an app `creator` role has `update` as one of their actions.

This matches my expectation.